xbmc/guilib/GUITexture.cpp

```diff
@@ -708,6 +721,20 @@ bool CGUITexture::SetAspectRatio(const CAspectRatio& aspect)
     return false;
 }
 
+void CGUITexture::SetTexture(
+    uint32_t width, uint32_t height, unsigned char* pixels, KD_TEX_FMT format, KD_TEX_ALPHA alpha)
+{
+  m_texture.Reset();
+  m_frameWidth = width;
+  m_frameHeight = height;
+
+  std::unique_ptr<CTexture> texture = CTexture::CreateTexture();
+  if (texture == nullptr || !texture->UploadFromMemory(width, height, 0, pixels, format, alpha))
```

Comment:
```suggestion
  if (!texture->UploadFromMemory(width, height, 0, pixels, format, alpha))
```
`CTexture::CreateTexture()` throws if the object can't be created but never returns a nullptr.